modules/core/lib/Auth/Process/AttributeAlter.php

```diff
@@ -175,13 +175,15 @@ public function process(&$request)
                     $attributes[$this->subject]
                 );
             } else {
+                /** @var array $attr */
+                $attr = $attributes[$this->subject];
                 $attributes[$this->target] = array_diff(
                     preg_replace(
                         $this->pattern,
                         $this->replacement,
-                        $attributes[$this->subject]
+                        $attr
```

Comment:
Psalm is smart enough to know that if arg3 is an array, the return type of preg_replace is an array as well..
It is however not smart enough to know that $attributes[$this->subject] is an array, so we have to annotate it.. Annotation inside the preg_replace doesn't work, so I had to do it earlier.

Reply:
Not currently.. A future (I guess 3.x) version of Psalm will support this even without annotation:
https://psalm.dev/r/41c55a6681
Let me find out the exact version and add it to the annotation so it can be removed in the future.

Reply:
Great!

Reply:
I gave up on this one, restored it, and just suppressed the issue..